src/extensions/default/Inline3DParametersEditor/main.js

```diff
@@ -0,0 +1,96 @@
+/*
+ * File based on InlineColorEditor/main.js
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var EditorManager           = brackets.getModule("editor/EditorManager"),
+        ExtensionUtils          = brackets.getModule("utils/ExtensionUtils"),
+        HTMLUtils               = brackets.getModule("language/HTMLUtils"),
+        Inline3dParametersUtils = require("Parameters3DUtils"),
+        Inline3DParameterEditor = require("Inline3DParameterEditor").Inline3DParameterEditor;
+
+    /**
+     * Prepare hostEditor for an Inline3DParameterEditor at pos if possible. Return
+     * editor context if so; otherwise null.
+     *
+     * @param {Editor} hostEditor
+     * @param {{line:Number, ch:Number}} pos
+     * @return {?{parameter:String, marker:TextMarker}}
+     */
+    function prepareParametersForProvider(hostEditor, pos) {
+        var ParameterRegex, tagStart, cursorLine, match, sel, start, end, endPos, marker, tagInfo;
+
+        sel = hostEditor.getSelection();
+        if (sel.start.line !== sel.end.line) {
+            return null;
+        }
+        var tagInfo = HTMLUtils.getTagInfo(hostEditor, sel.start);
+        var tag = tagInfo.tagName;
+        tagStart =Inline3dParametersUtils.TAG_START;
+        if(!tag.substr(0, 2) === tagStart) {
+            return null;
+        }
+
+        ParameterRegex = Inline3dParametersUtils.PARAMETERS_3D_REGEX;
+        ParameterRegex.lastIndex = 0;
+        tagInfo = HTMLUtils.getTagInfo(hostEditor, pos),
+        cursorLine = hostEditor.document.getLine(pos.line);
+        // Loop through each match of ParameterRegex and stop when the one that contains pos is found.
+        do {
+            match = ParameterRegex.exec(cursorLine);
```

Comment:
Actually, this looks wrong (or overly complicated). Why is this being done inside a loop when you're using a global regex. It seems that this should be outside the loop and evaluated only once. That will probably fix your odd regex uncertainty issue as well.
I think the code should look like:
```js
match = ParamRegex.exec(cursorLine);
if(!match) {
  return;
}

while(pos.ch < start || pos.ch > end) {
  start = match.index;
  end = start + match[0].index;
}
```
I also question the conditional here: `pos.ch < start || pos.ch > end`. What is it achieving? Why is it supposed to be less than `start` or greater than `end`? It's basically implying that `pos.ch` should NOT be between start and end which is weird. This at the very least needs some documentation.

cc @humphd